components/pluralrules/src/rules/ast.rs

```diff
@@ -403,11 +403,14 @@ pub struct SampleRange {
 /// use icu_pluralrules::rules::ast::*;
 /// DecimalValue {
 ///     integer: Value(1),
-///     decimal: Some(Value(5)),
+///     decimal: Some(DecimalNumber("5".to_string())),
 /// };
 /// ```
 #[derive(Debug, Clone, PartialEq)]
 pub struct DecimalValue {
     pub integer: Value,
-    pub decimal: Option<Value>,
+    pub decimal: Option<DecimalNumber>,
 }
+
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+pub struct DecimalNumber(pub String);
```

Comment:
this is the piece that I added to AST to support `5.00` without paying for it with the runtime AST (which uses `ast::Value(u64)`.

The tradeoff here is that I'm separating the fundamental node that stores value from the very similar node that stores a sample value, but I'm getting less of a hit on performance/memory usage at runtime.

I *don't* think that `String` is what I should have here, and I'm trying to see if `SmallStr` or `TinyStr` or `FixedDecimal` fits better here. The argument for `FixedDecimal` is that it's designed for storing numbers with precision. The argument for `TinyStr` is that it's designed to efficiently store small strings and samples are small. `TinyStr` is also `Copy` and likely the cheapest memory wise.

@sffc - do you have thoughts on it?

Reply:
`FixedDecimal` would be appropriate to represent the whole DecimalValue, both the integer and the fraction parts.  I personally think that would be nicest.  However, `FixedDecimal` could also be used for only the fraction part.  If you never do any math with it, though, storing it as a string seems to make sense.  In fact, you could store both the integer and the fraction parts as strings if you never do math with them.

`DecimalNumber` is not a very good name, because it implies that the type is something more than it is.  I think I have a preference for just putting the String directly into DecimalValue, like `pub decimal: Option<String>`.

Since this code shouldn't generally be executed at runtime, since the whole idea is to remove the sample strings, I don't think it matters whether you use `String`, `Cow<str>`, `SmallStr` or `TinyStr`.  If you use TinyStr, you'd probably want TinyStrAuto, unless you know for sure that the strings have bounded length.

Reply:
> In fact, you could store both the integer and the fraction parts as strings if you never do math with them.

Thanks!

> DecimalNumber is not a very good name,

Oh yeah, I don't like the name at all :) Thanks!

Ok, I'll look into this.

Reply:
I was able to use `String` with no benchmark impact, so, since its not runtime, I picked the side of fewer dependencies.